# HG changeset patch
# User sh@sh-laptop
# Date 1232640416 18000
#      Thu Jan 22 11:06:56 2009 -0500
# Node ID da95767511f557d7174cc702095c1565b58fc5da
# Parent  4a864c4e682de52e9b70b1f9c0886787202326a2
Remove descriptions of built-in variables from manual

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,12 @@
+2009-01-22  Soren Hauberg  <hauberg@gmail.com>
+
+        * interpreter/var.txi: Remove sections on built-in variables.
+
 2009-01-20  John W. Eaton  <jwe@octave.org>
 
 	* interpreter/system.txi (Filesystem Utilities): @DOCSTRING for
 	fstat.  Delete @DOCSTRING for lstat.  Add anchor for doc-lstat.
 
 2009-01-18  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
 	* interpreter/numbers.txi (Numeric Data Types): Trivial fixes.
diff --git a/doc/interpreter/var.txi b/doc/interpreter/var.txi
--- a/doc/interpreter/var.txi
+++ b/doc/interpreter/var.txi
@@ -50,29 +50,29 @@ documented internal variables and built-
 Case is significant in variable names.  The symbols @code{a} and
 @code{A} are distinct variables.
 
 A variable name is a valid expression by itself.  It represents the
 variable's current value.  Variables are given new values with
 @dfn{assignment operators} and @dfn{increment operators}.
 @xref{Assignment Ops, ,Assignment Expressions}.
 
-A number of variables have special built-in meanings.  For example,
-@code{ans} holds the current working directory, and @code{pi} names the
-ratio of the circumference of a circle to its diameter. @xref{Summary of
-Built-in Variables}, for a list of all the predefined variables.  Some
-of these built-in symbols are constants and may not be changed.  Others
-can be used and assigned just like all other variables, but their values
-are also used or changed automatically by Octave.
+There is one built-in variable with a special meaning. The @code{ans} variable
+always contains the result of the last computation, where the output wasn't
+assigned to any variable. The code @code{a = cos (pi)} will assign the value -1
+to the variable @code{a}, but will not change the value of @code{ans}. However,
+the code @code{cos (pi)} will set the value of @code{ans} to -1.
 
 Variables in Octave do not have fixed types, so it is possible to first
 store a numeric value in a variable and then to later use the same name
 to hold a string value in the same program.  Variables may not be used
 before they have been given a value.  Doing so results in an error.
 
+@DOCSTRING(ans)
+
 @DOCSTRING(isvarname)
 
 @DOCSTRING(genvarname)
 
 @DOCSTRING(namelengthmax)
 
 @menu
 * Global Variables::            
@@ -364,214 +364,8 @@ function allows this.
 Information about a function or variable such as its location in the
 file system can also be acquired from within Octave. This is usually
 only useful during development of programs, and not within a program.
 
 @DOCSTRING(type)
 
 @DOCSTRING(which)
 
-@node Summary of Built-in Variables
-@section Summary of Built-in Variables
-
-Here is a summary of all of Octave's built-in variables along with
-cross references to additional information and their default values.  In
-the following table @var{octave-home} stands for the root directory
-where all of Octave is installed (the default is @file{@value{OCTAVEHOME}},
-@var{version} stands for the Octave version number (for example,
-@value{VERSION}) and @var{arch} stands for the type of system for which
-Octave was compiled (for example, @code{x86_64-unknown-linux-gnu}).
-
-@vtable @code
-@item EDITOR
-@xref{Commands For History}.
-
-Default value: @code{"emacs"}.
-
-@item EXEC_PATH
-@xref{Controlling Subprocesses}.
-
-Default value: @code{":$PATH"}.
-
-@item OCTAVE_HOME
-
-Default value: @code{"@value{OCTAVEHOME}"}.
-
-@item PAGER
-@xref{Input and Output}.
-
-Default value: @code{"less", or "more"}.
-
-@item PS1
-@xref{Customizing the Prompt}.
-
-Default value: @code{"\s:\#> "}.
-
-@item PS2
-@xref{Customizing the Prompt}.
-
-Default value: @code{"> "}.
-
-@item PS4
-@xref{Customizing the Prompt}.
-
-Default value: @code{"+ "}.
-
-@item beep_on_error
-@xref{Errors and Warnings}.
-
-Default value: 0.
-
-@item completion_append_char
-@xref{Commands For Completion}.
-
-Default value: @code{" "}.
-
-@item default_save_options
-@xref{Simple File I/O}.
-
-Default value: @code{"ascii"}.
-
-@item crash_dumps_octave_core
-@xref{Simple File I/O}.
-
-Default value: 1.
-
-@item fixed_point_format
-@xref{Matrices}.
-
-Default value: 0.
-
-@item gnuplot_binary
-@xref{Three-Dimensional Plotting}.
-
-Default value: @code{"gnuplot"}.
-
-@item history_file
-@xref{Commands For History}.
-
-Default value: @code{"~/.octave_hist"}.
-
-@item history_size
-@xref{Commands For History}.
-
-Default value: 1024.
-
-@item ignore_function_time_stamp
-@xref{Function Files}.
-
-Default value: @code{"system"}.
-
-@item max_recursion_depth
-@xref{Recursion}.
-
-Default value: 256.
-
-@item output_max_field_width
-@xref{Matrices}.
-
-Default value: 10.
-
-@item output_precision
-@xref{Matrices}.
-
-Default value: 5.
-
-@item page_screen_output
-@xref{Input and Output}.
-
-Default value: 1.
-
-@item print_empty_dimensions
-@xref{Empty Matrices}.
-
-Default value: 1.
-
-@item save_precision
-@xref{Simple File I/O}.
-
-Default value: 17.
-
-@item saving_history
-@xref{Commands For History}.
-
-Default value: 1.
-
-@item sighup_dumps_octave_core
-@xref{Simple File I/O}.
-
-Default value: 1.
-
-@item sigterm_dumps_octave_core
-@xref{Simple File I/O}.
-
-Default value: 1.
-
-@item silent_functions
-@xref{Defining Functions}.
-
-Default value: 0.
-
-@item split_long_rows
-@xref{Matrices}.
-
-Default value: 1.
-
-@item struct_levels_to_print
-@xref{Data Structures}.
-
-Default value: 2.
-
-@item suppress_verbose_help_message
-@xref{Getting Help}.
-
-Default value: 1.
-@end vtable
-
-
-@node Defaults from the Environment
-@section Defaults from the Environment
-
-Octave uses the values of the following environment variables to set the
-default values for the corresponding built-in or internal variables.
-In addition, the values from the environment may be overridden by
-command-line arguments.  @xref{Command Line Options}.
-
-@vtable @code
-@item EDITOR
-@xref{Commands For History}.
-
-Built-in variable: @code{EDITOR}.
-
-@item OCTAVE_EXEC_PATH        
-@xref{Controlling Subprocesses}.
-
-Built-in variable: @code{EXEC_PATH}.
-Command-line argument: @code{--exec-path}.
-
-@item OCTAVE_PATH
-@xref{Function Files}.
-
-Internal variable changed by function @code{path}.
-Command-line argument: @code{--path}.
-
-@item OCTAVE_INFO_FILE
-@xref{Getting Help}.
-
-Internal variable changed by function @code{info_file}.
-Command-line argument: @code{--info-file}.
-
-@item OCTAVE_INFO_PROGRAM
-@xref{Getting Help}.
-
-Internal variable changed by function @code{info_program}.
-Command-line argument: @code{--info-program}.
-
-@item OCTAVE_HISTSIZE
-@xref{Commands For History}.
-
-Built-in variable: @code{history_size}.
-
-@item OCTAVE_HISTFILE
-@xref{Commands For History}.
-
-Built-in variable: @code{history_file}.
-@end vtable
